Tests/SwaggerParserTests/SwaggerParserTests.swift

```diff
@@ -1,20 +1,129 @@
+    }
+    
+    func testAllOfSupport() {
+        let url = testFixtureFolder.appendingPathComponent("test_all_of.json")
+        let jsonString = try! NSString(contentsOfFile: url.path, encoding: String.Encoding.utf8.rawValue) as String
+        let swagger = try! Swagger(JSONString: jsonString)
+        
+        guard
+            let baseDefinition = swagger.definitions.first(where: {$0.name == "TestAllOfBase"}),
+            case .object(let baseSchema) = baseDefinition.structure else {
+                XCTFail("TestAllOfBase is not an object schema."); return
         }
+        
+        SwaggerParserTests.validate(testAllOfBaseSchema: baseSchema)
+        swagger.definitions.validate(testAllOfChild: "TestAllOfFoo", withPropertyNames: ["foo"])
+        swagger.definitions.validate(testAllOfChild: "TestAllOfBar", withPropertyNames: ["bar"])
     }
+}
 
+/// MARK: Helper Functions
 
-    static var allTests = [
-        ("testUber", testUber),
-    ]
+fileprivate extension SwaggerParserTests {
+    /// Gets the base schema and child schema from a definition that defines an
+    /// `allOf` with one $ref (the base class) and one object schema.
+    class func getBaseAndChildSchemas(withDefinition definition: Structure<Schema>) -> (base: ObjectSchema?, child: ObjectSchema?) {
+        var base: ObjectSchema?
+        var child: ObjectSchema?
+        
+        if case .allOf(let schema) = definition.structure {
+            XCTAssertEqual(schema.schemas.count, 2)
+            
+            schema.schemas.forEach {
+                switch $0 {
+                case .object(let childSchema):
+                    child = childSchema
+                case .structure(let structure):
+                    guard case .object(let baseSchema) = structure.structure else {fatalError()}
+                    base = baseSchema
+                default: fatalError()
+                }
+            }
+        }
+        return (base: base, child: child)
+    }
+}
+
+/// MARK: Validation functions
+
+fileprivate extension SwaggerParserTests {
+    class func validate(testAllOfBaseSchema schema: ObjectSchema) {
+        validate(thatSchema: schema, named: "TestAllOfBase", hasRequiredProperties: ["base", "test_type"])
+    }
+
+    class func validate(thatSchema schema: ObjectSchema, named name: String, hasRequiredProperties properties: [String]) {
+        XCTAssertEqual(schema.properties.count, properties.count)
+        XCTAssertEqual(schema.required, properties)
+        
+        properties.forEach { property in
+            XCTAssertNotNil(schema.properties.first(where: {$0.key == property}))
+        }
+    }
+    
+    class func validate(thatParameter parameter: Parameter, named parameterName: String, isAnObjectNamed objectName: String, withPropertyName objectPropertyName: String) {
+        guard case .body(_, let schema) = parameter else {
+            XCTFail("\(parameterName) is not a .body."); return
+        }
```

Comment:
I like how you structured this with the series of guard statements.
A couple of nits:
- Newline after each closing curly preceding a statement.
- Let's avoid combining statements into one line like this - semicolons are the devil!
- XCTFail is a void function. You can just do:
```
return XCTFail(...)
```